js/views/property.js

```diff
@@ -378,6 +378,108 @@ app.views.property = function (accountNumber) {
     var taxBalanceUrl = 'http://www.phila.gov/revenue/realestatetax/?txtBRTNo=' + opa.parcel_number;
     app.hooks.taxBalanceLink.attr('href', taxBalanceUrl);
 
+    function updateTaxEstimate() {
+      // get assessment values
+      var marketValue = opa.market_value,
+          exemptValue = opa.exempt_building + opa.exempt_land;
+
+      // get selected tax rate
+      var rateVal = app.hooks.taxEstimateRate.val(),
+          useProposed = rateVal === 'proposed',
+          rate = useProposed ? 1.4572 : 1.3998;
+
+      // get homestead checkbox value
+      var homesteadCheckbox = app.hooks.taxEstimateHomesteadCheckbox,
+          isHomesteadChecked = homesteadCheckbox.prop('checked');
+
+      // check if they already have a homestead exemption
+      var existingHomesteadValue = opa.homestead_exemption || 0,
+          alreadyHasHomestead = existingHomesteadValue > 0;
+
+      // handle homestead exemption if necessary
+      if (isHomesteadChecked) {
+        // if we're using the proposed rate
+        if (useProposed) {
+          // and they already have an exemption
+          if (alreadyHasHomestead) {
+            // calculate their current homestead proportionate to 30k and
+            // apply to 45k
+            var proposedHomestead = existingHomesteadValue / 30000 * 45000;
+            // exemptValue += proposedHomestead;
+            exemptValue = proposedHomestead;
+          // if they don't already have a homestead, use 45k
+          } else {
+            exemptValue = 45000;
+          }
+        // if we're using the current rate
+        } else {
+          // use their existing homestead or the standard homestead of 30k
+          exemptValue = existingHomesteadValue || 30000;
+        }
+      }
+
+      // calculate tax estimate
+      var taxableValue = marketValue - exemptValue,
+          taxEstimate = taxableValue * rate / 100,
+          taxEstimateOrZero = Math.max(taxEstimate, 0);
+
+      // leaving this in here because it's super useful for debugging.
+      // console.log('taxable value', taxableValue, '=', 'marketValue', marketValue, '- exemptValue', exemptValue);
+      // console.log('tax estimate', taxEstimate, '= max of', taxEstimate, 'and 0');
+
+      // format and update
+      var formattedTaxEstimate = accounting.formatMoney(taxEstimateOrZero);
+      app.hooks.taxEstimateResult.text(formattedTaxEstimate);
+    }
+
+    function updateHomesteadLabel() {
+      // call foundation again to activate the homestead disabled tooltip
+      $(document).foundation();
+
+      var existingHomesteadValue = opa.homestead_exemption,
+          rateVal = app.hooks.taxEstimateRate.val(),
+          displayHomesteadValue;
+
+      // if they already have a homestead exemption, use that value
+      if (existingHomesteadValue > 0) {
+        if (rateVal === 'proposed') {
+          displayHomesteadValue = 45000;
```

Comment:
Shouldn't this be `Math.max(0, existingHomesteadValue / 30000 * 45000)`?

Reply:
Good catch, yes -- it should reflect the proportionate, proposed homestead. I'll push a change.